### ISSUE TITLE
Bot Instances Only Log To Files On Their First Initialization

### ISSUE THREAD

**Issue by breezko** [becurrie/titandash#537]
**Describe the bug**
If the bot detects a transition, and it doesnt get recovered, the bot terminates. After restarting it, the Bot won't write to the log file anymore.

**To Reproduce**
1. Get a transition error till termination.
2. restart bot in gui.
3. check log file timestamps/content.

**Expected Behavior**
Write to previous/new log file.

**Comment by becurrie** [comment]
@breezko  This is currently intended, since a bot "breaking" or hitting the transition limit is technically ending the session which makes sense.

This can be addressed to work as intended like it does now, while letting your TitanObserver implementation properly restart an instance and use the proper log and session (I think), I'll look into this and get something started at least.

**Comment by becurrie** [comment]
> @breezko This is currently intended, since a bot "breaking" or hitting the transition limit is technically ending the session which makes sense.
> 
> This can be addressed to work as intended like it does now, while letting your TitanObserver implementation properly restart an instance and use the proper log and session (I think), I'll look into this and get something started at least.

Scratch this, @breezko and I went over this, there is an issue here with the way logging is handled for new instances. See comment below.

**Comment by becurrie** [comment]
Currently, the `make_logger` function takes in an `instance` argument, this is fine, except that this information isn't unique enough to create unique loggers:

```python
_logger = logging.getLogger("{log_name}.{instance}".format(log_name=log_name, instance=instance.pk))
```

Since the logging module keeps these `getLogger` calls in memory, after the first instance is initialized, we don't ever properly add our handlers anymore because of this conditional check:

```python
_types = [type(handle) for handle in _logger.handlers]
if logging.FileHandler not in _types:
    _logger.addHandler(file_handler)
if logging.StreamHandler not in _types:
    _logger.addHandler(console_handler)
if TitanBotLoggingHandler not in _types:
    _logger.addHandler(socket_handler)
```

See the linked pull request to see the fixes for this.

**Comment by becurrie** [comment]
Resolved by #538
